src/components/Purchase/CheckoutPage.js

```diff
@@ -21,34 +22,40 @@ const total = 140.0;
 
 const CheckoutPage = (/*{ album, quantity, subtotal, tax, total}*/) => {
   const [clientSecret, setClientSecret] = useState('');
-
+  const [orderData, setOrderData] = useState(null);
```

Comment:
Not needed as comonent state, since we only use the orderData inside useEffect

Reply:
Instead, I would recommend moving te `location.state` extraction out of useEffect and putting here:

```suggestion
  // Extract order data from navigation.state
  const { orderData } = location.state || {};
```

that way, it'll be easier to pass the order data.total value into the CheckoutForm component